python/lsst/ap/association/association.py

```diff
@@ -307,7 +315,7 @@ def match(self, dia_objects, dia_sources, score_struct):
         # Argwhere returns a array shape (N, 1) so we access the index
         # thusly to retrieve the value rather than the tuple.
         for (src_idx,) in np.argwhere(np.logical_not(used_dia_source)):
-            associated_dia_object_ids[src_idx] = dia_sources[src_idx].getId()
+            associated_dia_object_ids[src_idx] = dia_sources[int(src_idx)].getId()
```

Comment:
Why is `src_idx` ever something other than an int?

Reply:
The argwhere returns numpy.int type for src_idx. For some reason lsst.afw.table.SourceCatalog doesn't like the numpy type as an index hence the wrapping in the int().

Reply:
Hmm... I can't find any documentation for `numpy.int`, but if it's the same as `numpy.int_` (which is C++ `long`) then that would explain it. Though as long as `used_dia_source` has less than 2 billion elements you should be fine.